_includes/specials/taproot/ja/03-p2wpkh-to-p2tr.md

```diff
@@ -0,0 +1,105 @@
+  P2WPKHと同じため、トランザクションのシリアライゼーションの変更について心配する必要はありません。
+
+- **署名メッセージの作成:** これは使用トランザクションのデータに対するコミットメントです。
+  データのほとんどは、P2WPKHトランザクションに署名するのと同じものですが、
+  フィールドの順番が[変更され][BIP341 sigmsg]、いくつかの追加項目が署名されています。
+  これを実装するのは、さまざまなデータをハッシュしシリアライズするだけなので、
+  コードを書くのは簡単です。
+
+- **署名メッセージのハッシュに署名:** Schnorr署名を作成するには、さまざまなな方法があります。
+  最善の方法は、「独自の暗号を使う」のではなく、信頼できる十分レビューされたライブラリの関数を使用することです。
+  ただ、何らかの理由によりそれができない場合は、
+  [BIP340][]は、ECDSA署名を作成するためのプリミティブが利用可能であれば、
+  簡単に実装できるアルゴリズムを提供します。署名ができたら、
+  インプットのwitnessデータに入れ、使用トランザクションを送信します。
+
+ブロック{{site.trb}}でTaprootがアクティベートされる前でも、
+testnetやパブリックなデフォルト[signet][topic signet]、
+Bitcoin Coreのプライベートなregtestモードを使ってコードをテストできます。
+オープンソースのウォレットにTaprootのサポートを追加する場合、
+他の開発者があなたのコードから学べるように、
+Bitcoin Wikiの[taproot uses][wiki taproot uses]ページや
+[bech32m adoption][wiki bech32 adoption]ページにその実装のPRへのリンクを追加することをお勧めします。
+
+[^electrum-segwit]:
+    Electrumがsegwit v0にアップグレードされた際、
+    bech32アドレスで受け取りたい人は誰でも新しいシードを生成する必要がありました。
+    これは技術的には必要ありませんでしたが、
+    Electrumの作者は自分たちのカスタムシードの導出方法にいくつかの新しい[機能][electrum seeds]を導入することができました。
+    その１つがシードのバージョン番号で、シードを使用するスクリプトを指定する機能です。
+    これにより古いスクリプトを安全に非推奨にできます（例えば、将来リリースされるElectrumのバージョンでは、
+    従来のP2PKHアドレスへの受信がサポートされなくなる可能性があります）。
+
+    Electrumの開発者がバージョン付きのシードを展開していたのと同時期に、
+    Bitcoin Coreの開発者は、[output script descriptor][topic descriptors]を使用して、
+    （他の問題の解決に加えて）スクリプトの非推奨を可能にするという同じ問題を解決し始めました。
+    次の表は、Electrumのバージョン付きのシードとBitcoin Coreのdescriptorを、
+    以前両方のウォレットで使用され、
+    現在も多くの他のウォレットで一般的に使用されている*implicit scripts*方式と比較したものです。
+
+    <table>
+      <tr>
+        <th>スクリプト管理</th>
+        <th>初期バックアップ</th>
+        <th>新しいスクリプトの導入</th>
+        <th>スキャン（帯域幅/CPUコスト）</th>
+        <th>非推奨スクリプト</th>
+      </tr>
+
+      <tr>
+        <th markdown="1">Implicit scripts (例：[BIP44][])</th>
+        <td>シードワード</td>
+        <td>自動（ユーザー操作は不要）</td>
+        <td>サポートされているすべてのスクリプトをスキャン、O(n)</td>
+        <td>サポートされていないスクリプトを使用していることをユーザーに警告する方法はありません</td>
+      </tr>
+
+      <tr>
+        <th>Explicit scripts（バージョン付きシード）</th>
+        <td>シードワード（バージョンビットを含む）</td>
+        <td>ユーザーは新しいシードのバックアップが必要。
+        資金は2つの別々のウォレットに分割されるか、ユーザーは旧ウォレットから新しいウォレットに資金を送信する必要があります</td>
+        <td>単一のスクリプトテンプレートのみをスキャン、O(1)</td>
+        <td>サポートされていないスクリプトに関するユーザーへの警告</td>
+      </tr>
+
+      <tr>
+        <th markdown="1">Explicit scripts ([descriptor][topic descriptors])</th>
+        <td>シードワードとdescriptor</td>
+        <td>ユーザーは新しいdescriptorのバックアップが必要</td>
+        <td>実際に使用されたスクリプトテンプレートのみをスキャン、O(n); 新しいウォレットの場合はn=1</td>
+        <td>サポートされていないスクリプトに関するユーザーへの警告</td>
+      </tr>
+    </table>
+
+{% include linkers/issues.md issues="" %}
+[electrum seeds]: https://electrum.readthedocs.io/en/latest/seedphrase.html#motivation
+[bip340 alt signing]: https://github.com/bitcoin/bips/blob/master/bip-0340.mediawiki#alternative-signing
+[bip341 cite22]: https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki#cite_note-22
+[`tr()` descriptors]: /en/preparing-for-taproot/#taproot-descriptors
```

Comment:
this link can be /ja/preparing-for-taproot/#taproot-descriptors

Reply:
oh, thanks! I updated.